### ISSUE TITLE
Loading external HTML or image as banner

### ISSUE THREAD

**Issue by rica-carv** [e107inc/e107#1491]

**Comment by nesjett** [comment]
I dont catch you, how were you doing this on v1.4? as I can see you create a campaing and asign an image to it.

There is no option for adding urls, html´s nor code, maybe are you referencing to adding html to a custom menu?

Grts,
- Nes

**Comment by rica-carv** [comment]
@nesjett I could do that in V1.4, by assigning on the banner image field (wich was editable text on v1.4) a link to a html file, instead of a image file.
Now on V2.1 you can only add images, since that field in no more editable as text....
We can't even link to an external image (as it was possible with v1.4), since we can only use images already present on the media manager...

**Comment by nesjett** [comment]
In my v1.4 versión i dont have such a text área, just an imagen picker (radio buttons) so, or your e107 or mine are bugged heheh

**Comment by LaocheXe** [comment]
@nesjett there are some ways to do that, for example change it in the database. I never missed with the banner plugin a whole lot, but I remember a few things.

@rica-carv can you send me the banner plugin files you got? forbiddenchaos@gmail.com

**Comment by rica-carv** [comment]
@LaocheXe The one's i'm using right now are the same that are here on github, on the banner plugin...
Or you want the v1.4 ones?

**Comment by nesjett** [comment]
@LaocheXe yes, I can figure out how to do that, but I dont remember it by default on e107

**Comment by Moc** [comment]
@rica-carv What is the purpose of having an HTML page as an image? To be able to update it externally/ In that case, I agree that an external image link is something to consider, this would also take care of the need for an HTML page then (which I don't think is ever going to be supported).

**Comment by rica-carv** [comment]
@nesjett @LaocheXe The way i was doing it on v1.4, was to have the html file on the images folder, under the banner plugin, and point the image field of a specific banner to the html file. Just that, and it worked like a charm....
@Moc Since V2.x has somehow "lost" this "abillity", i've decided to convert all my original html banners to images, so.....

**Comment by Moc** [comment]
@rica-carv I am trying to understand the use of the HTML file. What was the purpose of it? Maybe we can make some improvement in the banner plugin. For example, I think that the banner plugin should support external images. That's why I re-opened this issue.

**Comment by rica-carv** [comment]
@Moc Sorry for closing this. I was primary using the HTML to make a banner from a image smaller, and also using plain text on it, for [instance ](http://www.selospt.tk/e107_images/banners/firefox_lonelyspooky.html)(check the source)

**Comment by MikeyGMT** [comment]
Many affiliate programmes give you the html code to use in you banner. This function was supported in previous versions of Banner.

**Comment by rica-carv** [comment]
@MikeyGMT Indeed, but now the only way is to have a image for the banner, no HTML...

**Comment by CaMer0n** [comment]
I think we'll need another field for html.

**Comment by rica-carv** [comment]
Or try to get back to the old v1.x way.
If you have a field for HTML and another for the image, it might get some conflicts, for instance, if you have already loaded a image inside the HTML....

**Comment by CaMer0n** [comment]
The old v1.x way has no way to enter raw HTML code from what I can see.

**Comment by rica-carv** [comment]
@CaMer0n No, it doesn't, but instead of the image path, you could have a path to a HTML file that had its banner rendered, ie:
`<img src='https://raw.github.com/jlongster/mozilla.org/master/contribute/buttons/468x60arrow_b_pt.png' alt='Firefox Download Button' border='0' />`
